ftw/file/tests/test_download.py

```diff
@@ -94,6 +94,30 @@ def test_inline_download(self, browser):
         self.assertEquals('inline; filename="file.doc"',
                           browser.headers['content-disposition'])
 
+    @browsing
+    def test_head_request(self, browser):
+        browser.default_driver = LIB_TRAVERSAL
+
+        browser.login()
+
+        # Accessing "@@download" on the file would normally result in a 302,
+        # telling us to got to "@@download/file/file.doc" instead.
+        # But since the testbrowser follows the redirects, we get a 200 instead.
+        browser.open(self.context, view='@@download', method='HEAD')
+        self.assertEqual(200, browser.status_code)
+        self.assertEqual(
+            'http://nohost/plone/f1/@@download/file/file.doc',
+            browser.url
+        )
+
+        # Open the correct url used to access the file.
+        browser.login().open(self.context, view='@@download/file/file.doc', method='HEAD')
```

Comment:
I'd leave the `login()` since that has already happened.

Reply:
I have removed it since the login happens earlier.